### PR TITLE
Prevent tests from polluting git dir with fourtitwo.json files

### PR DESCRIPTION
**Pull request by fingolfin**

### CODE REVIEW

File: src/Serialization/main.jl [hunk 1]
```diff
@@ -220,9 +220,9 @@ See [`load`](@ref).
 # Examples
 
 ```jldoctest
-julia> save("fourtitwo.json", 42);
+julia> save("/tmp/fourtitwo.json", 42);
 
-julia> load("fourtitwo.json")
+julia> load("/tmp/fourtitwo.json")
```
Comment (thofma) [inline src/Serialization/main.jl:225]: I guess many people don't have `/tmp`. Will this be a problem? I mean, when running the doctests locally.

**Comment by fieker** [comment]
On Wed, Oct 26, 2022 at 10:57:07PM -0700, Tommy Hofmann wrote:
> @thofma commented on this pull request.
> 
> 
> 
> >  
> -julia> load("fourtitwo.json")
> +julia> load("/tmp/fourtitwo.json")
> 
> I guess many people don't have `/tmp`. Will this be a problem?
Who? Where?
I'd say any *nix should have it - unless OSX does not. WSL supports it
as well...
Othewise its a system call gettmp or similar..

**Comment by benlorenz** [comment]
Could we do something like `cd(mktempdir(cleanup=false))` in the doctestsetup-block, or will this break other stuff?

**Comment by HereAround** [comment]
Minor side comment - It seems that https://github.com/oscar-system/Oscar.jl/pull/1642 was only a temporary fix. Thus, you may want to revert (or modify) this change here.

**Comment by thofma** [comment]
I forgot that we are *nix only. Everything is fine then.

**Comment by fingolfin** [comment]
Yeah I was *intentionally* a bit lazy here by using `/tmp`, my rationale was this: I didn't want to complicate the examples by adding code to query for a temporary directory. At the same time, all systems we support should have `/tmp`, plus it only really matters for people who run the doctests, i.e., our developers and some Github CI jobs.

That said, if someone wants to implement a better solution (e.g. add a setup block that is not visible to the user and which changes into a temporary directory, executes the test there, then changes back -- if such a thing is possible?) then by all means go for it :-)

**Comment by fingolfin** [comment]
> Could we do something like `cd(mktempdir(cleanup=false))` in the doctestsetup-block, or will this break other stuff?

It may be safe, but it could lead to weird heisenbugs down the road if we later add tests which expect to be executed in a certain directory (as those tests would work in isolation, but not in a full doctest run -- depending on the order in which we process things). That said, it probably is just wrong for tests to have such a behavior, so perhaps we should just do it the way you propose anyway? Dunno...

**Review by benlorenz — APPROVED** [review]
I tried my suggestion (in a few different places) but unfortunately that doesn't work well with Documenter.jl, there are various `cd(<some relative path>)` calls in the code that fail when the directory is changed.

So I think we can merge this now.

**Pull request merged**